core/src/main/java/tc/oc/pgm/destroyable/DestroyableFactory.java

```diff
@@ -33,15 +56,17 @@ public DestroyableFactory(
       double destructionRequired,
       @Nullable ImmutableSet<Mode> modeList,
       boolean showProgress,
-      boolean sparks,
-      boolean repairable) {
+      String sparks,
+      boolean repairable,
+      Element el)
+      throws InvalidXMLException {
     super(id, name, required, showOptions, owner, proximityMetric);
     this.region = region;
     this.materials = materials;
     this.destructionRequired = destructionRequired;
     this.modeList = modeList;
     this.showProgress = showProgress;
-    this.sparks = sparks;
+    this.sparks = SparksType.fromString(sparks, el);
```

Comment:
parsing should not occur on the constructor, the constructor should take in the already parsed attributes.

Additionally we have XMLUtils to parse enums, you likely didn't need to add your own fromString